### DISCUSSION TITLE
Link account with oauth providers.

### DISCUSSION THREAD

**Discussion by alfredjophy** [pocketbase/pocketbase#3750] [Q&A] [answered]
TLDR;

Link currently logged in user with auth providers like google, discord, steam ( using OIDC) etc.

Setup - I have manual code exchange setup with sveltekit.

When I log in to an oauth provider while logged in, I end up creating a new account. 
I am aware of automatic email linking but I dont want that, as steam doesnt return an email field.

I have seen similar discussion which mentioned logging in with username and password and using the same client. I am not sure how that would work with manual code exchange.

**Comment by ganigeorgiev** [comment]
_This is kindof similar to https://github.com/pocketbase/pocketbase/discussions/3737#discussioncomment-7536173._

The "all-in-one" flow internally still uses the "Manual code exchange", just the redirect handler is builtin and its response is returned through a realtime `@oauth2` subscription, or in other words it is effectively the same.

It is not clear what exactly you are asking, but if you want to "link" an OAuth2 user with existing PocketBase user, it is enough the SDK client to be authenticated with your user (_email or username, it doesn't matter as long as there is `Authorization: TOKEN` header_) before sending the OAuth2 auth request.

Note that this type of linking will work only if the OAuth2 user is not already associated to another PocketBase user.

Alternatively, if you want to perform the linking based on some more advanced criteria, you could check the example with the hook in the related discussion.

**Reply by alfredjophy** [reply]
Yeah I did see the other issue.

I believe the issue is with how the oauth redirect works.
I have the pb_auth cookie stored and have setup a hook to auth with `loadFromCookie`.
For some reason, the /oauth/redirect that I have made doesnt get the pb_auth cookie.

I believe we can close this issue as pocketbase is not the cause.

Thanks

**Comment by alfredjophy** [comment, marked as answer]
I have solved the issue. 
Setting sameSite=lax fixes the issue.

I am not sure if this is a good way.
